lms/djangoapps/instructor/views/api.py

```diff
@@ -3058,15 +3056,11 @@ def generate_certificate_exceptions(request, course_id, generate_for=None):
             status=400
         )
     else:
-        students = User.objects.filter(
-            id__in=users,
-            certificatewhitelist__course_id=course_key,
-            certificatewhitelist__whitelist=True
-        )
+        students = 'whitelisted_not_generated'
 
     if students:
         # generate certificates for students if 'students' list is not empty
```

Comment:
`students` is no longer a list. It's also confusing, in this stretch of code, that `students` can refer to either a "list" of one student ID or a string describing a list of students that we eventually want to query for. This variable should be renamed at the very least to `student_set`, and below (L3272), `student` should be renamed to `specific_student` (and should not be a list)